library/Vanilla/AddonManager.php

```diff
@@ -359,7 +371,15 @@ private function scanAddonDirs($type) {
         foreach ($this->scanDirs[$type] as $subdir) {
             $paths = glob(PATH_ROOT."$subdir/*", GLOB_ONLYDIR | GLOB_NOSORT);
             foreach ($paths as $path) {
-                $result[basename($path)] = substr($path, $strlen);
+                $basename = basename($path);
+                if (!array_key_exists($basename, $result)) {
+                    $result[basename($path)] = substr($path, $strlen);
```

Comment:
$result[$basename] = substr($path, $strlen);
instead of 
$result[basename($path)] = substr($path, $strlen);